source/standalone/workflows/ray/isaac_ray_tune.py

```diff
@@ -0,0 +1,345 @@
+# Copyright (c) 2022-2024, The Isaac Lab Project Developers.
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+import argparse
+import importlib.util
+import os
+import sys
+from time import sleep
+
+import isaac_ray_util
+import ray
+from ray import air, tune
+from ray.tune.search.optuna import OptunaSearch
+from ray.tune.search.repeater import Repeater
+
+"""
+This script breaks down an aggregate tuning job, as defined by a hyperparameter sweep configuration,
+into individual jobs (shell commands) to run on the GPU-enabled nodes of the cluster.
+By default, (unless combined as a sub-job in a resource-wrapped aggregate job), one worker is created
+for each GPU-enabled node in the cluster for each individual job.
+
+Each hyperparameter sweep configuration should include the workflow,
+runner arguments, and hydra arguments to vary.
+
+This assumes that all workers in a cluster are homogeneous. For heterogeneous workloads,
+create several heterogeneous clusters (with homogeneous nodes in each cluster),
+then submit several overall-cluster jobs with :file:`../submit_isaac_ray_job.py`.
+KubeRay clusters on Google GKE can be created with :file:`../launch.py`
+
+To report tune metrics on clusters, a running MLFlow server with a known URI that the cluster has
+access to is required. For KubeRay clusters configured with :file:`../launch.py`, this is included
+automatically, and can be easily found with with :file:`grok_cluster_with_kubectl.py`
+
+Usage:
+
+.. code-block:: bash
+
+    ./isaaclab.sh -p source/standalone/workflows/ray/isaac_ray_tune.py -h
+
+    # Examples
+    # Local
+    ./isaaclab.sh -p source/standalone/workflows/ray/isaac_ray_tune.py --run_mode local \
+    --cfg_file source/standalone/workflows/ray/hyperparameter_tuning/vision_cartpole_cfg.py \
+    --cfg_class CartpoleRGBNoTuneJobCfg
+
+    # Remote (run grok cluster or create config file mentioned in :file:`submit_isaac_ray_job.py`)
+    ./isaaclab.sh -p source/standalone/workflows/ray/submit_isaac_ray_job.py \
+    --aggregate_jobs isaac_ray_tune.py # cfg_file is relative to ray folder
+    --cfg_file hyperparameter_tuning/vision_cartpole_cfg.py \
+    --cfg_class CartpoleRGBNoTuneJobCfg --mlflow_uri <MFLOW_URI_FROM_GROK_OR_MANUAL>
+
+"""
+
+docker_prefix = "/workspace/isaaclab/"
+base_dir = os.path.expanduser("~")
+python_exec = "./isaaclab.sh -p"
+workflow = "source/standalone/workflows/rl_games/train.py"
+num_workers_per_node = 1  # needed for local parallelism
```

Comment:
Now that you moved these to the top of the file they are constants so should be all caps